core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -1931,11 +1935,25 @@ public void doUpdateRoleListOfUser(String userName, String[] deletedRoles, Strin
             }
 
             if (newRoles != null && newRoles.length > 0) {
+
+                ArrayList<String> newRoleList = new ArrayList<>();
+                for (String role : newRoles) {
+                    if(!isExistingRole(role)){
+                        String errorMessage = "The role: " + role + " does not exist.";
+                        throw new UserStoreException(errorMessage);
+                    }
+                    if (!isUserInRole(userName, role)) {
+                        newRoleList.add(role);
+                    }
+                }
+
+                String[] rolesToAdd = newRoleList.toArray(new String[newRoleList.size()]);
                 // if user name and role names are prefixed with domain name,
                 // remove the domain name
+                RoleBreakdown breakdown = getSharedRoleBreakdown(rolesToAdd);
```

Comment:
The comment above and the method does not match.
The method seems to be actually separating about shared roles from normal roles

Reply:
Fixed with da7df85bc577b726fb7556a6e76ee398e11b3f80